test/test_nn.py

```diff
@@ -9810,26 +9810,28 @@ def helper(n, c, h, w, kernel_size, stride=None,
         helper(10, 512, 31, 31, 3, stride=2)
         helper(1, 129, 8, 8, 3, stride=2)
 
-    @onlyCUDA
     def test_max_pool2d(self, device):
-        def helper(n, c, h, w, ks):
-            x = torch.randn(n, c, h, w, device='cuda', dtype=torch.float, requires_grad=True)
+        def helper(n, c, h, w, ks, requires_grad):
+            x = torch.randn(n, c, h, w, device=device, dtype=torch.float, requires_grad=requires_grad)
             ref_x = x.detach().clone().cpu().requires_grad_()
 
-            pool = torch.nn.MaxPool2d(kernel_size=ks)
-
-            y = pool(x)
-            ref_y = pool(ref_x)
+            pool = torch.nn.MaxPool2d(kernel_size=ks, return_indices=True)
```

Comment:
Would you elaborate on this change? In particular:

- In the original return_indices was not set and it default to False. 
- Doesn't your change only affect the return_indices = False codepath?